modules/foreman_host.py

```diff
@@ -0,0 +1,217 @@
+        org = self._entities.Architecture(self._server, name=name, **params)
+        response = org.search(set(), {'search': 'name={}'.format(name)})
+
+        if len(response) == 1:
+            return response[0]
+        else:
+            self._module.fail_json(msg="No architecture found for %s" % name)
+
+    def find_organization(self, name, **params):
+        org = self._entities.Organization(self._server, name=name, **params)
+        response = org.search(set(), {'search': 'name={}'.format(name)})
+
+        if len(response) == 1:
+            return response[0]
+        else:
+            self._module.fail_json(msg="No organization found for %s" % name)
+
+    def find_location(self, name, **params):
+        loc = self._entities.Location(self._server, name=name, **params)
+        response = loc.search(set(), {'search': 'name={}'.format(name)})
+
+        if len(response) == 1:
+            return response[0]
+        else:
+            self._module.fail_json(msg="No Location found for %s" % name)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            server_url=dict(required=True),
+            username=dict(required=True, no_log=True),
+            password=dict(required=True, no_log=True),
+            verify_ssl=dict(required=False, type='bool', default=False),
+            hostname=dict(required=True, no_log=False),
+            mac=dict(required=True, no_log=False),
+            organization=dict(required=True, no_log=False),
+            location=dict(required=True, no_log=False),
+            architecture=dict(required=True, no_log=False),
+            operatingsystem=dict(required=True, no_log=False),
+        ),
+        supports_check_mode=True
+    )
+
+    if not HAS_NAILGUN_PACKAGE:
+        module.fail_json(msg="Missing required nailgun module (check docs or install with: pip install nailgun")
+
+    server_url = module.params['server_url']
+    username = module.params['username']
+    password = module.params['password']
+    hostname = module.params['hostname']
+    mac = module.params['mac']
+    organization = module.params['organization']
+    location = module.params['location']
+    architecture = module.params['architecture']
+    operatingsystem = module.params['operatingsystem']
+    verify_ssl = module.params['verify_ssl']
+
+    server = ServerConfig(
+        url=server_url,
+        auth=(username, password),
+        verify=verify_ssl
+    )
+    ng = NailGun(server, entities, module)
+
+    # Lets make an connection to the server with username and password
+    try:
+        org = entities.Organization(server)
+        org.search()
+    except Exception as e:
+        module.fail_json(msg="Failed to connect to Foreman server: %s " % e)
+
+    try:
+        ng.create_host(hostname, mac, organization, location, architecture, operatingsystem)
+    except Exception as e:
+        module.fail_json(msg=to_native(e))
+
+    module.exit_json(changed=True, result="Host")
```

Comment:
`changed` should not always be `True`, please have a look how you can make `changed` dynamic

examples are https://github.com/theforeman/foreman-ansible-modules/blob/master/modules/katello_manifest.py and https://github.com/theforeman/foreman-ansible-modules/blob/master/modules/katello_lifecycle_environment.py